Silicon/CommonSocPkg/Library/HeciLib/HeciLib.c

```diff
@@ -182,17 +184,38 @@ HeciGetFwCapsSkuMsg (
   EFI_STATUS                      Status;
   UINT32                          Length;
   GEN_GET_FW_CAPSKU              *MsgGenGetFwCapsSku;
-  GEN_GET_FW_CAPS_SKU_ACK        *MsgGenGetFwCapsSkuAck;
+  GEN_GET_FW_CAPS_SKU_ACK_DATA   *MsgGenGetFwCapsSkuAck;
+  LOADER_GLOBAL_DATA             *LdrGlobal;
+  ME_BIOS_PAYLOAD                *MbpDataHob;
+  UINT32                          MbpDataHobLen;
+  UINT8                          *DataPtr;
 
   if ((MsgGetFwCaps == NULL) || (MsgGetFwCapsAck == NULL) ) {
     return EFI_INVALID_PARAMETER;
   }
 
   MsgGenGetFwCapsSku    = (GEN_GET_FW_CAPSKU *)MsgGetFwCaps;
-  MsgGenGetFwCapsSkuAck = (GEN_GET_FW_CAPS_SKU_ACK *)MsgGetFwCapsAck;
+  MsgGenGetFwCapsSkuAck = (GEN_GET_FW_CAPS_SKU_ACK_DATA *)MsgGetFwCapsAck;
 
   //
-  // Allocate MsgGenGetFwVersion data structure
+  // Fill the FW capapbility data.
+  //
+  LdrGlobal = (LOADER_GLOBAL_DATA *)GetLoaderGlobalDataPointer ();

```

Comment:
Can you implement a  GetFspHobListPtr() function in BootloaderCorePkg\Library\BootloaderLib\BootloaderLib.c ?
I saw lots of places we use FSP hob, so it makes sense to abstract it to provide a API function.